JOSESwift/Sources/JWEHeader.swift

```diff
@@ -272,6 +297,44 @@ extension JWEHeader: CommonHeaderParameterSpace {
             parameters["crit"] = newValue
         }
     }
+
+    /// Header Parameters Used for ECDH Key Agreement - Ephemeral Public Key
+    public var epk: ECPublicKey? {
+        get {
+            guard let jwkParameters = parameters["epk"] as? [String: String] else {
+                return nil
+            }
+
+            guard let json = try? JSONEncoder().encode(jwkParameters) else {
```

Comment:
In one of my tests, I noticed that `[String: String]` is too restrictive. I'm creating a JWE using Web Crypto and for the `epk` value, I'm using its `exportKey` function. Here is a quick and simple example code that you can test in a browser's console:

```js
await crypto.subtle.exportKey('jwk', (await crypto.subtle.generateKey({name: 'ECDH', namedCurve: 'P-384'}, false, ['deriveBits'])).publicKey)
```

It will output something like this:

```json
{
    "crv": "P-384",
    "ext": true,
    "key_ops": [],
    "kty": "EC",
    "x": "klRAjXE1X-VamQCmarKMnFxmsRRqNhIBGY6wS8cTL-Gjji_R_NkBg3EoVF55Aqmn",
    "y": "KiqMj4boroqWJTHrY91QzGZe1cTfrjzYiHOsjZdjNYX_6FVoXEvPKkGsw__2sbPX"
}
```

My suggestion would be to encode the `epk` header param less restrictive, e.g., like this:

```suggestion
            guard let jwkParameters = parameters["epk"] else {
                return nil
            }

            guard let json = try? JSONSerialization.data(withJSONObject: jwkParameters) else {
```

What do you think?